### DISCUSSION TITLE
updating password and email in pocketbase

### DISCUSSION THREAD

**Discussion by Alanl101** [pocketbase/pocketbase#2311] [Q&A]
Hello pocketable seems to be very vague. I can update my username, name and other fields with ease I believe I should be to update my email like my normal fields but I am more concerned with updating the password. So I believe my logic is correct I request for the password reset and then I confirm it by passing over the new password to the await function in the form of a string. as a note I'd like to mention I am following the api details for request password and confirm password for javascript. Thank you!

**Comment by ganigeorgiev** [comment]
I'm not sure that I understand what exactly is the question.

You can change an auth record password 2 ways:
- request a password reset (_this doesn't require the user to be authenticated_)
- change the password via the `update("RECORD_ID", { ... })` call by providing `oldPassword`, `password`, `passwordConfirm` fields (_this depends on the Update API rule of the auth collection; as noted in the documentation, if you are authenticated as admin or the user has Manage API rule access, then `oldPassword` is not required_).

The email can be changed by regular users only if you send `requestEmailChange()` request (_note that currently [there is a limitation](https://github.com/orgs/pocketbase/projects/2?pane=issue&itemId=22401932) related to the accounts created with OAuth2_).

If the above doesn't help, please elaborate and provide a code sample what you are trying to do.
